lib/libutils/isoc/arch/arm/setjmp_a64.S

```diff
@@ -94,4 +94,6 @@ BTI(	bti	c)
 	br	x30
 	.size	longjmp, .-longjmp
 
+#ifndef __clang__
```

Comment:
This will effectively disable BTI for the whole TA when Clang is used, won't it? I'd rather have something more explicit in `mk/config.mk` (refuse to set `CFG_TA_BTI` when `COMPILER==clang`).
Or better: can't we work out something with the definition of `setjmp()`? If we make it a macro that expands to inline assembly: `br setjmp; bti j`, wouldn't it work?

Reply:
This will disable BTI only for TA's which call function setjmp or longjmp so warning is required only if setjmp or longjmp is being used from TA with clang. So refusing to set CFG_TA_BTI when compiler is clang for TA's where setjmp/longjmp() is not used is not required.

Reply:
> This will disable BTI only for TA's which call function setjmp or longjmp

I have not made any experiment, but yes I see how this can be true when libutils is an archive (static library). The linker will eliminate the unused object file from the archive. However, when libutils is a shared library (`CFG_ULIBS_SHARED=y`), I think the whole library will lack the BTI feature bit.

Reply:
You are right. the .so file won't have the property.
There are implementations where ret is used in longjmp instead of br. Should we explore using those instead of one from newlib ?

Reply:
Ah yes that would be great if we could workaround this issue relatively easily...

Reply:
libc has similar impleentation as what we picked from new libc
https://github.com/embeddedartistry/libc/blob/master/arch/aarch64/src/longjmp.s

In TF-A
https://elixir.bootlin.com/arm-trusted-firmware/latest/source/lib/libc/aarch64/setjmp.S#L58
They are using ret here.

Android seems to use ret
https://android.googlesource.com/platform/bionic/+/808d176e7e0dd727c7f929622ec017f6e065c582/libc/arch-arm64/bionic/setjmp.S
(They have done other things like mangling of pointers etc)

Clearly in the implementation we have br x30 is used to avoid some mis-predictions. So changing setjmp/longjmp implementation is probably not the right way to go.

Reply:
> Clearly in the implementation we have br x30 is used to avoid some mis-predictions. So changing setjmp/longjmp implementation is probably not the right way to go.

It it's just a preformance issue that might be acceptable, we could simply do:
```C
#ifdef __clang__
	// <insert comment about Clang setjmp bug>
	ret
#else
	// use br not ret, as ret is guaranteed to mispredict
	br x30
#endif
```

Reply:
Updated in review commit